examples/apps/zoom-app/pages/run_inference.py

```diff
@@ -0,0 +1,43 @@
+import psutil
+import numpy as np
+import streamlit as st
+
+from sys import platform
+from landingai.pipeline.image_source import NetworkedCamera
+from landingai.predict import Predictor
+
+
+is_win = platform == "win32"
+
+
+if "api_key" in st.session_state and "endpoint_id" in st.session_state:
+    webcam_on = False
+    if is_win:
+        from webcam_detect import WebcamDetect
+
+        webcam_detect = WebcamDetect()
+        webcam_on = webcam_detect.is_active_app("zoom")
+
+    if (
+        len([p for p in psutil.process_iter() if "zoom" in p.name().lower()]) > 0
+        and webcam_on
+    ):
+        model = Predictor(
+            st.session_state["endpoint_id"], api_key=st.session_state["api_key"]
+        )
+        video_src = NetworkedCamera(0, fps=1)
```

Comment:
A few minor comments here:

- With 1 FPS, I think the end user might reach rate limit
- There is a new `Webcam` class that might be more intuitive for end users in this case. It's basically a wrapper on top of `NetworkedCamera`.
- Image sources are now context managers, meaning that you can use them with `with` statement. So, at the end of the `with` block, the webcam will be released. This is a bit safer, to make sure the camera will be released before the end of the script lifetime. Like this:

```py
with Webcam(fps=1) as video_src:
    for frame in vide_src:
        ...
```

Reply:
Nice! I like this, I will update the code

Reply:
Hey, just to be clear: it closes the webcam resources when it was the script itself opening that resource.

I see that you removed your "zoom" check, for example. I don't think our implementation will really close the webcam resource from Zoom in this case. It is more likely that we will fail to acquire access to the webcam if it is already in use by another process.

Reply:
I'm aware, that was a totally different feature I wanted to remove.

Reply:
As thiago says customers may hit the rate limiter with 1 FPS (it is set to 40 images per min)... I would change it to
`Webcam(fps=1, motion_detection_threshold=10)`

This way only changes are evaluated

Reply:
I changed it to 0.5 FPS, adding the motion_detection_threshold kind of ruins the demo because it's setup to track how often you are looking/not looking at the camera (so if you are still and it only triggers with motion detection you wont get how often you are holding a position and looking at the camera).

The demo runs at around 1 prediction every ~2-3 seconds already because of the delay from the call to the predict API.